### PR TITLE
SC-5644 - a11y, skip links for navigation

### PR DESCRIPTION
**Pull request by haniapalac**
# Description

## Links to Tickets or other pull requests

https://ticketsystem.schul-cloud.org/browse/SC-5644

## Changes

Changed meta tags, added skip links to main landmarks so the user, when tabbing, doesn't need to tab through the whole sidebar etc. The links are visually hidden and visible only when focused.

## Screenshots of UI changes

image
image

## Approval for review
- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definition of Done
More and detailed information on the *definition of done* can be found [on Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 599
- Added 14
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5880626&bid=19248363)

File: locales/de.json [hunk 1]
```diff
@@ -905,6 +905,24 @@
 	},
 	"global": {
 		"topMenu": {
+			"ariaLabel": "topbar",
+			"dropdown": {
+				"alertButton": {
+					"title": "Es gibt ein Problem"
+				},
+				"fullscreenButton": {
+					"title": "Fullscreen"
```
Comment (luizzaly) [inline locales/de.json:914]: "Vollbild" / "Vollbildmodus" to keep it in german

File: locales/de.json [hunk 2]
```diff
@@ -905,6 +905,24 @@
...
+				},
+				"QRCode": {
+					"title": "QR-Code"
+				},
+				"notifications": {
+					"title": "Benachrichtigungen"
+				},
+				"currentUser": {
+					"ariaLabel": "Current user:"
```
Comment (luizzaly) [inline locales/de.json:923]: please change to "aktueller Benutzer" to keep it german

File: locales/de.json [hunk 3]
```diff
@@ -1020,6 +1039,11 @@
 			"tuesday": "Dienstag",
 			"wednesday": "Mittwoch"
 		},
+		"skipLinks": {
+			"mainContentSkipLink": "Skip to main content",
```
Comment (luizzaly) [inline locales/de.json:1043]: "Hauptbereich"

File: locales/de.json [hunk 4]
```diff
@@ -1020,6 +1039,11 @@
...
+			"topNavbarSkipLink": "Skip to top navbar",
```
Comment (luizzaly) [inline locales/de.json:1044]: "Navigation"

File: locales/de.json [hunk 5]
```diff
@@ -1020,6 +1039,11 @@
...
+			"footerSkipLink": "Skip to footer"
```
Comment (luizzaly) [inline locales/de.json:1045]: "Footer"

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 599
- Added 14
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5880626&bid=19248363)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 599
- Added 15
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5880626&bid=19248363)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 29
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5880626&bid=19248363)

File: locales/de.json [hunk 6]
```diff
@@ -1158,8 +1159,31 @@
...
+				"QRCode": {
+					"title": "QR-Code"
+				}
+			},
 			"qrHintText": "Lotse mit dem QR-Code andere Nutzer:innen auf diese Seite"
 		},
+		"skipLinks": {
+			"mainContentSkipLink": "Hauptbereich",
+			"topNavbarSkipLink": "Navigation",
+			"footerSkipLink": "Footerc"
```
Comment (dominikbruechner) [inline locales/de.json:1185]:
```suggestion
			"footerSkipLink": "Footer"
```

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 29
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5880626&bid=19248363)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 29
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5880626&bid=19248363)

**Comment by haniapalac** [comment]
Now the skip links should be properly styled for both logged in and logged out users:
image
image
image
image

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 29
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5880626&bid=19248363)

File: theme/brb/views/lib/default.hbs [hunk 7]
```diff
@@ -61,6 +62,11 @@
 </head>
 
 <body class="{{#if inline}}window-inline{{/if}}">
+<dl class="skipLinks" role="navigation" aria-label="skip links">
+	<dd><a href="#main-content">{{ $t "global.skipLinks.mainContentSkipLink" }}</a></dd>
+	<dd><a href="#top-navbar">{{ $t "global.skipLinks.topNavbarSkipLink" }}</a></dd>
+	<dd><a href="#footer">{{ $t "global.skipLinks.footerSkipLink" }}</a></dd>
+</dl>
```
Comment (luizzaly) [inline theme/brb/views/lib/default.hbs:69]: I think a skip link for the sidebar might be useful as well..?

Comment (dominikbruechner) [reply theme/brb/views/lib/default.hbs:69]: I think sidebar is the first element you navigation to after the skip links

File: locales/de.json [hunk 8]
```diff
@@ -1125,7 +1126,7 @@
 			}
 		},
 		"sidebar": {
-			"ariaLabel": "sidebar",
+			"ariaLabel": "Sidebar",
```
Comment (dominikbruechner) [inline locales/de.json:1129]:
```suggestion
			"ariaLabel": "Seitenleiste",
```

File: locales/de.json [hunk 9]
```diff
@@ -1108,7 +1108,8 @@
 			"brokenPen": "zerbrochener Stift"
 		},
 		"label": {
-			"room": "Raum"
+			"room": "Raum",
+			"skipLinks": "Skip links"
```
Comment (dominikbruechner) [inline locales/de.json:1112]:
```suggestion
			"skipLinks": "Links Überspringen"
```

File: locales/de.json [hunk 10]
```diff
@@ -1175,11 +1177,34 @@
...
+				},
+				"QRCode": {
+					"title": "QR-Code"
+				}
+			},
 			"qrHintText": "Lotse mit dem QR-Code andere Nutzer:innen auf diese Seite"
+		},
+		"skipLinks": {
+			"mainContentSkipLink": "Hauptbereich",
+			"topNavbarSkipLink": "Navigation",
```
Comment (dominikbruechner) [inline locales/de.json:1203]:
```suggestion
			"topNavbarSkipLink": "Navigation oben",
```

**Pull request merged**